compiler/rustc_mir/src/interpret/eval_context.rs

```diff
@@ -927,7 +931,12 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
     #[must_use]
     pub fn generate_stacktrace(&self) -> Vec<FrameInfo<'tcx>> {
         let mut frames = Vec::new();
-        for frame in self.stack().iter().rev() {
+        for frame in self
+            .stack()
+            .iter()
+            .rev()
+            .skip_while(|frame| frame.instance.def.requires_caller_location(*self.tcx))
+        {
```

Comment:
AFAIK this is also used for Miri error messages... do we really want those backtraces to skip frames? That sounds rather confusing. When we get a panic at runtime, with `RUST_BACKTRACE` we also show the full trace, not just the non-caller-location frames, right?

Reply:
Even during CTFE, while this makes sense for panics, note that this logic is sued for *all* CTFE errors. I am not sure if that's always what we want.

Reply:
oh.. right. we should limit this to panics

Reply:
Turns out this is also the source of some Miri panics in our diagnostics code which didn't expect pruned backtraces.

If we want this kind of pruning, we should not to it in the lowest-level `generate_stacktrace` function.